src/main/java/appeng/crafting/CraftingTreeNode.java

```diff
@@ -50,14 +53,13 @@
 	private final IAEItemStack what;
 	// what are the crafting patterns for this?
 	private final ArrayList<CraftingTreeProcess> nodes = new ArrayList<CraftingTreeProcess>();
-	int bytes = 0;
-	boolean canEmit = false;
-	boolean cannotUse = false;
-	long missing = 0;
-	long howManyEmitted = 0;
-	boolean exhausted = false;
-
-	boolean sim;
+	private int bytes = 0;
+	private boolean canEmit = false;
+	private boolean cannotUse = false;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "cannotUse" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)